Source/CombatExtended/CombatExtended/Loadouts/Utility_HoldTracker.cs

```diff
@@ -301,18 +301,18 @@ static public bool GetAnythingForDrop(this Pawn pawn, out Thing dropThing, out i
             {
                 storage.Add(pawn.equipment.Primary.def, new Integer(1));
                 gun = pawn.equipment.Primary.TryGetComp<CompAmmoUser>();
-                if (gun != null && gun.UseAmmo)
+                if (gun != null && gun.UseAmmo && gun.CurrentAmmo != null)
                     storage.Add(gun.CurrentAmmo, new Integer(gun.CurMagCount));
             }
             // get the pawn's inventory
             foreach (Thing thing in pawn.inventory.innerContainer)
             {
                 Thing thing2 = thing.GetInnerIfMinified();
-                if (!storage.ContainsKey(thing2.def))
+                if (thing2 != null && !storage.ContainsKey(thing2.def))
```

Comment:
Makes no sense to check thing2 for null here, but leave it unchecked before storage[thing2.def]. Use a check 'if (thing2 == null) continue;' or something before checking both instead.